message_definitions/v1.0/common.xml

```diff
@@ -4716,27 +4716,28 @@
       <field type="float[4]" name="quaternion">Quaternion of the sensor orientation in vehicle body frame (w, x, y, z order, zero-rotation is 1, 0, 0, 0). Zero-rotation is along the vehicle body x-axis. This field is required if the orientation is set to MAV_SENSOR_ROTATION_CUSTOM. Set it to 0 if invalid."</field>
     </message>
     <message id="133" name="TERRAIN_REQUEST">
-      <description>Request for terrain data and terrain status</description>
+      <description>Request for terrain data and terrain status. See terrain protocol docs: https://mavlink.io/en/services/terrain.html</description>
       <field type="int32_t" name="lat" units="degE7">Latitude of SW corner of first grid</field>
       <field type="int32_t" name="lon" units="degE7">Longitude of SW corner of first grid</field>
       <field type="uint16_t" name="grid_spacing" units="m">Grid spacing</field>
       <field type="uint64_t" name="mask" display="bitmask" print_format="0x%07x">Bitmask of requested 4x4 grids (row major 8x7 array of grids, 56 bits)</field>
     </message>
     <message id="134" name="TERRAIN_DATA">
-      <description>Terrain data sent from GCS. The lat/lon and grid_spacing must be the same as a lat/lon from a TERRAIN_REQUEST</description>
+      <description>Terrain data sent from GCS. The lat/lon and grid_spacing must be the same as a lat/lon from a TERRAIN_REQUEST. See terrain protocol docs: https://mavlink.io/en/services/terrain.html</description>
       <field type="int32_t" name="lat" units="degE7">Latitude of SW corner of first grid</field>
       <field type="int32_t" name="lon" units="degE7">Longitude of SW corner of first grid</field>
       <field type="uint16_t" name="grid_spacing" units="m">Grid spacing</field>
       <field type="uint8_t" name="gridbit">bit within the terrain request mask</field>
       <field type="int16_t[16]" name="data" units="m">Terrain data MSL</field>
     </message>
     <message id="135" name="TERRAIN_CHECK">
+      <deprecated since="2020-08" replaced_by="">Deprecated (without replacement) as is not part of terrain protocol: https://mavlink.io/en/services/terrain.html.</deprecated>
```

Comment:
this is used by mavproxy, why deprecate it?

Reply:
Hi @tridge Because when I asked on ardupilot-gcs@googlegroups.com no one responded that it was needed

```
I just updated MAVLink Terrain protocol docs. The main sequence is correct for ArduPilot based on it's interaction with QGC.

QGC does not use TERRAIN_CHECK and it isn't needed for the GCS to know the vehicle terrain status. Therefore I made a (possibly incorrect) assumption that no one uses it and explicitly noted that it is not part of the protocol. 

Does anyone here support TERRAIN_CHECK? If so I would revert that statement.
```
How is it used? The terrain protocol as per QGC and Mission Planner is as documented here: https://mavlink.io/en/services/terrain.html
They indicated that they could get the map without need for this check.

Reply:
@hamishwillee It is used to implement the "Terrain Check" popup menu item (right click on map) in mavproxy. It allows the user to query the status of the terrain data at an arbitrary location